### PR TITLE
improve addline + updateline form submission

### PR DESCRIPTION
**Pull request by cfoellmann**
- after addline = success -> clear $_POST + redirect to <url>#addline (jump to addline form again)

- after updateline = success -> clear $_POST + keep redirect to updated line

### CODE REVIEW

File: htdocs/comm/propal/card.php [hunk 1]
```diff
@@ -1163,38 +1163,9 @@
...
-					unset($_POST['date_startday']);
-					unset($_POST['date_startmonth']);
-					unset($_POST['date_startyear']);
-					unset($_POST['date_endhour']);
-					unset($_POST['date_endmin']);
-					unset($_POST['date_endsec']);
-					unset($_POST['date_endday']);
-					unset($_POST['date_endmonth']);
-					unset($_POST['date_endyear']);
+					$urltogo = preg_replace('/--IDFORBACKTOPAGE--/', $object->id, $urltogo) . '#addline';
```
Comment (eldy) [inline htdocs/comm/propal/card.php:1166]: I am not a big fan of making redirect when it is not absolutely necessary. It increase the "flash" effect on slow servers, making a bad experience for end user.
Why not instead, introducing a function unsetPostAfterInsertLine() ?

**Comment by cfoellmann** [comment]
in the current state of the code the page does NOT jump down to the addline form after submission! Bad UX

Are you sure that this header location does create an additional pageload? I am pretty sure that it does not. Otherwise the unset of POST would not work for the user!

**Comment by eldy** [comment]
Yes, i am sure that adding a header with a redirect create an overload, because it returns a result to the browser to say to the browser to ask another page. So there is on answer and one request more on the network.

To jump to the addline form (and retreive same position than before submit), we must use a generic feature:
- add the class "reposition" on the CSS of the button (so page position is restored at same place that before clicking "submit". It is better than using the #abc (position differs a little bit from position of button).
- It works magically if the css was added on a href link. But if it was added on a button of a form, you must also check that a hidden field called "page_y" is inside the form and it will work too.
`print '<input type="hidden" name="page_y" value="" />';`

**Comment by eldy** [comment]
This PR is now too old, so i close it. A lot of code may have been validated since the PR and the PR may be obsolete, or too difficult to merge due to other changes. Best choice is too cancel it to avoid to keep PR with too low chances to be validated.

**Comment by cfoellmann** [comment]
I will try to validate and create a new PR for anything that is still needed

**Pull request closed without merging**